### PR TITLE
feat(domeventobserver): #16412 add support for usePassive option to the DomEventObserver

### LINKED ISSUE #16412: Add support for `usePassive` option to the `DomEventObserver`
## 📝 Provide a description of the new feature

Currently, `DomEventObserver` doesn't pass the `usePassive` option to the listener -https://github.com/ckeditor/ckeditor5/blob/5248c7be58ba444cc060d7b4ce7047376e0ccd1d/packages/ckeditor5-engine/src/view/observer/domeventobserver.ts#L73-L78. 

This feature request is about adding support for `usePassive` option in the `DomEventObserver`.

---

If you'd like to see this feature implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by timblack1-WC**
Fixes #16412

This change permits a user to create a `scrollend` event listener which does not block future scroll events.

### CODE REVIEW

**Comment by timblack1-WC** [comment]
Would this change require changing the documentation also?  If so, let me know where and I will consider whether I can help change the documentation.

**Comment by timblack1-WC** [comment]
I have a signed copy of the CLA; how can I submit it to you?

File: packages/ckeditor5-engine/src/view/observer/domeventobserver.ts [hunk 1]
```diff
@@ -56,6 +56,12 @@ export default abstract class DomEventObserver<
 	 */
 	public useCapture: boolean = false;
 
+	/**
+	 * If set to `true`, indicates that the function specified by listener will never call preventDefault()
```
Comment (niegowski) [inline packages/ckeditor5-engine/src/view/observer/domeventobserver.ts:60]:
```suggestion
	 * If set to `true`, indicates that the function specified by listener will never call `preventDefault()`.
```

**Review by niegowski — COMMENTED** [review]
Please add tests.

**Comment by Mati365** [comment]
I'm closing in favor of https://github.com/ckeditor/ckeditor5/pull/17098. I added tests and proper changelog entry there.

**Pull request closed without merging**